doc/reference/reference_sql/sql_statements_and_clauses.rst

```diff
@@ -496,95 +496,185 @@ Data types may also appear in :ref:`CAST <sql_function_cast>` functions.
-    -- this is an example of an attempted primary-key violation
-    -- (the third INSERT will fail because 55, 'a' is a duplicate)
-    CREATE TABLE t3 (s1 INTEGER, s2 STRING, PRIMARY KEY (s1, s2));
-    INSERT INTO t3 VALUES (55, 'a');
-    INSERT INTO t3 VALUES (55, 'b');
-    INSERT INTO t3 VALUES (55, 'a');
-
-PRIMARY KEY plus AUTOINCREMENT modifier may be specified in one of two ways: |br|
-- In a column definition after the words PRIMARY KEY, as in ``CREATE TABLE t (c INTEGER PRIMARY KEY AUTOINCREMENT);`` |br|
-- In a PRIMARY KEY (column-list) after a column name, as in ``CREATE TABLE t (c INTEGER, PRIMARY KEY (c AUTOINCREMENT));`` |br|
-When AUTOINCREMENT is specified, the column must be a primary-key column and it must be INTEGER or UNSIGNED. |br|
+**Example 2: two-column primary key**
+
+1.  Create a ``book`` table with the primary key defined on two columns:
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: create_book_table_start
+        :end-before: create_book_table_end
+        :dedent:
+
+    Insert data into this table:
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_books_start
+        :end-before: insert_books_end
+        :dedent:
+
+2.  On an attempt to add the existing book, the following error is raised:
+
+    ..  literalinclude:: /code_snippets/test/sql/primary_key_table_constraint_test.lua
+        :language: sql
+        :start-after: insert_duplicate_book_start
+        :end-before: insert_duplicate__book_end
+        :dedent:
+
+PRIMARY KEY with the AUTOINCREMENT modifier may be specified in one of two ways:
+
+-   In a column definition after the words PRIMARY KEY, as in ``CREATE TABLE t (c INTEGER PRIMARY KEY AUTOINCREMENT);``
+
+-   In a PRIMARY KEY (column-list) after a column name, as in ``CREATE TABLE t (c INTEGER, PRIMARY KEY (c AUTOINCREMENT));``
+
+When AUTOINCREMENT is specified, the column must be a primary-key column and it must be INTEGER or UNSIGNED.
+
 Only one column in the table may be autoincrement.
 However, it is legal to say ``PRIMARY KEY (a, b, c AUTOINCREMENT)`` -- in that case, there
-are three columns in the primary key but only the first column (``a``) is AUTOINCREMENT.
+are three columns in the primary key but only the third column (``c``) is AUTOINCREMENT.
 
 As the name suggests, values in an autoincrement column are automatically incremented.
 That is: if a user inserts NULL in the column, then the stored value will be the smallest
 non-negative integer that has not already been used.
 This occurs because autoincrement columns are associated with :doc:`sequences </reference/reference_lua/box_schema_sequence/create_index>`.
 
-UNIQUE constraints look like this: |br|
-:samp:`UNIQUE ({column-name} [, {column-name}...])`
+UNIQUE constraints look like this:
```

Comment:
```suggestion
**UNIQUE constraints**

UNIQUE constraints look like this:
```
I think it's necessary to separate the paragraphs about PRIMARY | UNIQUE | CHECK constraints from each other somehow (with **bold**?). It's a bit difficult to figure out where a new paragraph starts.

Reply:
Agree, will do